PensionManagementPensionerService/Models/Repository/Implementation/GuardianRepository.cs

```diff
@@ -1,32 +1,62 @@
-﻿using PensionManagementPensionerService.Models.Repository.Interfaces;
+﻿using Microsoft.EntityFrameworkCore;
+using PensionManagementPensionerService.Models.Context;
+using PensionManagementPensionerService.Models.Repository.Interfaces;
 
 namespace PensionManagementPensionerService.Models.Repository.Implementation
 {
     public class GuardianRepository : IGuardianRepository
     {
-        public Task<GuardianDetails> AddGuardian(GuardianDetails guardianDetails)
+        private readonly AppDbContext _appDbContext;
+
+        public GuardianRepository(AppDbContext appDbContext)
+        {
+            this._appDbContext = appDbContext;
+        }
+        public async Task<GuardianDetails> AddGuardian(GuardianDetails guardianDetails)
         {
-            throw new NotImplementedException();
+            var result = await _appDbContext.GuardianDetails.AddAsync(guardianDetails);
```

Comment:
Don't we have a method that takes the input directly and save it in DB instead of adding it separately and then saving the changes?